actors/builtin/miner/miner_actor.go

```diff
@@ -2014,14 +2024,26 @@ func (a Actor) WithdrawBalance(rt Runtime, params *WithdrawBalanceParams) *abi.T
 		// Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
 		// and repay fee debt now.
 		feeToBurn = RepayDebtsOrAbort(rt, &st)
-	})
 
-	amountWithdrawn := big.Min(availableBalance, params.AmountRequested)
-	builtin.RequireState(rt, amountWithdrawn.GreaterThanEqual(big.Zero()), "negative amount to withdraw: %v", amountWithdrawn)
-	builtin.RequireState(rt, amountWithdrawn.LessThanEqual(availableBalance), "amount to withdraw %v < available %v", amountWithdrawn, availableBalance)
+		// To get the actual amount to be withdrawed
+		amountWithdrawn = big.Min(availableBalance, params.AmountRequested)
+		builtin.RequireState(rt, amountWithdrawn.GreaterThanEqual(big.Zero()), "negative amount to withdraw: %v", amountWithdrawn)
+		builtin.RequireState(rt, amountWithdrawn.LessThanEqual(availableBalance), "amount to withdraw %v < available %v", amountWithdrawn, availableBalance)
+		if info.Beneficiary != info.Owner {
+			remainingQuota := info.BeneficiaryTerm.Available(rt.CurrEpoch())
+			builtin.RequirePredicate(rt, remainingQuota.GreaterThan(big.Zero()), exitcode.ErrForbidden,
```

Comment:
I would like to go fail-fast principle. And from saving network resource point of view, I would suggest both places use `>` since transferring 0 doing nothing but taking computing resource on each node.